### PR TITLE
retrieve albamgr client config in more cases

### PR DESCRIPTION
**Pull request by toolslive**

### CODE REVIEW

**Comment by domsj** [comment]
We should probably fix the (same) bug in https://github.com/openvstorage/alba/blob/master/ocaml/src/remotes.ml#L41 too? Or at least not pretend that we're being clever there?

Btw nice test suite addition

File: ocaml/src/proxy_server.ml [hunk 1]
```diff
@@ -480,35 +484,71 @@ let proxy_protocol (alba_client : Alba_client.alba_client)
...
+type refresh_result =
+  | Retry
+  | Res of Albamgr_protocol.Protocol.Arakoon_config.t
+
 let refresh_albamgr_cfg
     ~loop
     albamgr_client_cfg
     (alba_client : Alba_client.alba_client)
     destination =
+  let retrieve_from_any_node_except () =
```
Comment (domsj) [inline ocaml/src/proxy_server.ml]: I think you meant to drop the `_except` suffix

**Comment by toolslive** [comment]
currently there are quite a few `loose cannon` tests. I plan to aggregate them in a suite (xml results, jenkins,...) in the immediate future.

File: ocaml/src/albamgr_client.ml [hunk 2]
```diff
@@ -509,10 +512,50 @@ let wrap_around (ara_c:Arakoon_client.client) =
...
+      | node_name :: rest ->
+         Lwt.catch
+           (fun  () ->
+            let node_cfg  =
+              Hashtbl.find node_hashtbl node_name
+            in
+            let open Albamgr_protocol.Protocol.Arakoon_config in
+            let cfg = from_node_client_cfg node_cfg in
+            Lwt_log.debug_f "retrieving from %s" node_name >>= fun () ->
+            Client_helper.with_client'
```
Comment (domsj) [inline ocaml/src/albamgr_client.ml]: At some point in the past we considered it possible that a node would accept a connection and not do anything with it for a long time (due to some arakoon bug at the time?), which is why we added https://github.com/openvstorage/arakoon/commit/fe49c8b764130307a4384c5215d366fb8d5bc06b to arakoon. I suggest doing the same thing here to protect us from such a scenario.

Comment (toolslive) [inline ocaml/src/albamgr_client.ml]: we were investigating a but and the addition was done under that hypothesis. IIRC.

**Pull request merged**